ClientCore/ProfanityFilter.cs

```diff
@@ -75,16 +68,21 @@ private static string StarCensoredMatch(Match m)
 
         private string ToRegexPattern(string wildcardSearch)
         {
-            string regexPattern = Regex.Escape(wildcardSearch);
-            regexPattern = regexPattern.Replace(@"\*", ".*?");
-            regexPattern = regexPattern.Replace(@"\?", ".");
-            if (regexPattern.StartsWith(".*?"))
-            {
-                regexPattern = regexPattern.Substring(3);
-                regexPattern = @"(^\b)*?" + regexPattern;
-            }
-            regexPattern = @"\b" + regexPattern + @"\b";
-            return regexPattern;
+            string pattern = Regex.Escape(wildcardSearch).Replace(@"\*", ".*?").Replace(@"\?", ".");
+
+            bool startsWithWildcard = wildcardSearch.StartsWith("*") || wildcardSearch.StartsWith("?");
+            bool endsWithWildcard = wildcardSearch.EndsWith("*") || wildcardSearch.EndsWith("?");
+
+            if (startsWithWildcard && endsWithWildcard)
+                return pattern; 
+            
+            if (startsWithWildcard)
+                return pattern + @"\b";
+
+            if (endsWithWildcard)
+                return @"\b" + pattern;
+
+            return @"\b" + pattern + @"\b";
```

Comment:
Can you explain the changes here and prove this method returns the same result as if this PR were not merged?